TA_dataset/bin/dataset_api.py

```diff
@@ -131,14 +127,27 @@ def ds_lrq_run_loop(
                 last_step_seen=steps_done,
                 forward_tag=forward_tag,
             )
+            logger().debug(response)
             result = response.parsed
-            steps_done = result.steps_completed
+            if result:
+                steps_done = result.steps_completed
+                retry = 0
+            else:
+                retry += 1
```

Comment:
Eventually it would be nice to abstract retries into the client library itself :)

Reply:
My observation is, that when it gets to the 500 status, it's not resolved by these subsequent calls. So it could be most likely removed.

I will add comment.